ros2_java_desktop.repos

```diff
@@ -1,28 +1,28 @@
 repositories:
   ament/ament_java:
     type: git
-    url: https://github.com/ros2-java/ament_java.git
-    version: master
+    url: https://github.com/jacobperron/ament_java.git
+    version: dsv_file_templates
```

Comment:
what is different in this branch?
I was using upstream master

Reply:
It adds DSV style environment hooks (https://github.com/jacobperron/ament_java/commit/b36a59bef4e81040f058c020adf57ce3271851a2).

I think the `foxy-devel` branch of `ros2_java` depends on this patch (perhaps we could relax the dependency), hence the build failure [here](https://github.com/osrf/ros2_java/runs/981416625):

```
CMake Error at /opt/ros/foxy/share/ament_cmake_core/cmake/index/ament_index_get_resource.cmake:62 (message):
    ament_index_get_resource() called with not existing resource ('templates'
    'ament_build_type_gradle_classpath_dsv')
  Call Stack (most recent call first):
    /home/runner/work/ros2_java/ros2_java/ros_ws/install/ament_cmake_export_jars/share/ament_cmake_export_jars/cmake/ament_export_jars.cmake:58 (ament_index_get_resource)
    CMakeLists.txt:147 (ament_export_jars)
```

I'm surprised that you are able to compile `rcljava_common` without that patch.

Reply:
> I'm surprised that you are able to compile rcljava_common without that patch.

I'm sorry you're right, I'm using that branch.